spec/controllers/api/orders_controller_spec.rb

```diff
@@ -0,0 +1,160 @@
+
+      context 'as a regular user' do
+        before do
+          allow(controller).to receive(:spree_current_user) { regular_user }
+          get :index
+        end
+
+        it "returns unauthorized" do
+          assert_unauthorized!
+        end
+      end
+
+      context 'as an admin user' do
+        before do
+          allow(controller).to receive(:spree_current_user) { admin_user }
+          get :index
+        end
+
+        it "retrieves a list of orders with appropriate attributes,
+            including line items with appropriate attributes" do
+
+          returns_orders(json_response)
+        end
+
+        it "formats completed_at to 'yyyy-mm-dd hh:mm'" do
+          completed_dates = json_response['orders'].map{ |order| order['completed_at'] }
+          correct_formats = completed_dates.all?{ |a| a == order1.completed_at.strftime('%B %d, %Y') }
+
+          expect(correct_formats).to be_truthy
+        end
+
+        it "returns distributor object with id key" do
+          distributors = json_response['orders'].map{ |order| order['distributor'] }
+          expect(distributors.all?{ |d| d.key?('id') }).to be_truthy
+        end
+
+        it "returns the order number" do
+          order_numbers = json_response['orders'].map{ |order| order['number'] }
+          expect(order_numbers.all?{ |number| number.match("^R\\d{5,10}$") }).to be_truthy
+        end
+      end
+
+      context 'as an enterprise user' do
+        context 'producer enterprise' do
+          before do
+            allow(controller).to receive(:spree_current_user) { supplier.owner }
+            get :index
+          end
+
+          it "does not display line items for which my enterprise is a supplier" do
+            assert_unauthorized!
+          end
+        end
+
+        context 'coordinator enterprise' do
+          before do
+            allow(controller).to receive(:spree_current_user) { coordinator.owner }
+            get :index
+          end
+
+          it "retrieves a list of orders" do
+            returns_orders(json_response)
+          end
+        end
+
+        context 'hub enterprise' do
+          before do
+            allow(controller).to receive(:spree_current_user) { distributor.owner }
+            get :index
+          end
+
+          it "retrieves a list of orders" do
+            returns_orders(json_response)
+          end
+        end
+      end
+
+      context 'with pagination' do
+        before do
+          allow(controller).to receive(:spree_current_user) { distributor.owner }
+        end
+
+        it 'returns pagination data when query params contain :per_page]' do
+          get :index, per_page: 15, page: 1
+
+          expect(json_response['pagination']).to eq pagination_data
+        end
+      end
+    end
+
+    private
+
+    def returns_orders(response)
+      keys = response['orders'].first.keys.map(&:to_sym)
+      expect(order_attributes.all?{ |attr| keys.include? attr }).to be_truthy
+    end
```

Comment:
I'm not sure this is worth the extra complexity. Isn't enough to check for an `id` field in `response['orders']`?

Reply:
This was part of migrating tests from one area to another. Checking all the attributes are there means we know the serializer is returning what it should do, but it also makes this test more coupled with the code. Should I cut it?

Reply:
I'm okay with leaving it as the work is done already.

Reply:
It's just the I'm always very concerned about performance, complexity and changeability of the test suite as it tends to grow a lot. But your call, it's not a big deal.